src/tardev-snapshotter/src/snapshotter.rs

```diff
@@ -82,42 +98,152 @@ impl Store {
     fn new(root: &Path) -> Self {
         Self {
             root: root.into(),
-            signatures: None,
+            signatures: Some(<HashMap<String, LayerInfo>>::new()),
         }
     }
 
-    fn lazy_read_signatures(&mut self) -> Result<()> {
-        if self.signatures == None {
-            debug!("Loading signatures");
-            self.read_signatures()
-                .context("Failed to read signatures")?;
+    async fn lazy_read_signatures(&mut self, image_name: String) -> Result<()> {
```

Comment:
Would it make sense to do this backwards? If signatures were not loaded yet, load the file (init scenario) and then we can separately add a new signatures from registry. We could probably also refactor further to just load the signatures on startup (instead of lazily) and then just add the new signature from registry when need be.

Reply:
Yup, it makes sense to load signatures for the approved container image list on startup, these container images could be hosted in MCR or GHCR where we can not push signatures to. Then load signatures of the images from registry, these images can be signed by the trusted key at any time instead of the build time.